### PR TITLE
[WIP] Add support for polymorphic associations.

### PR DESCRIPTION
**Pull request by beauby**
Properly choose the association key for polymorphic associations.

### CODE REVIEW

**Comment by joaomdmoura** [comment]
I know this seems a `[WIP]` but I'll point it out just to make sure we don't forget: tests are missing

File: lib/active_model/serializer/reflection.rb [hunk 1]
```diff
@@ -78,6 +78,17 @@ def build_association(subject, parent_serializer_options)
 
       private
 
+      def polymorphic_key(assoc_value)
+        reflection_key = assoc_value.class.model_name
+        assoc_value.respond_to?(:each) ? reflection_key.plural : reflection_key.singular
```
Comment (joaomdmoura) [inline lib/active_model/serializer/reflection.rb]: :kissing_heart: what I need, I'm merging this in a branch of mine :grin:

Comment (beauby) [inline lib/active_model/serializer/reflection.rb]: Cool, tell me whether this worked! (usage is `has_one :addressable, polymorphic: true`)

Comment (joaomdmoura) [inline lib/active_model/serializer/reflection.rb]: works like a charm ;)

File: lib/active_model/serializer/reflection.rb [hunk 2]
```diff
@@ -78,6 +78,17 @@ def build_association(subject, parent_serializer_options)
...
+      end
+
+      def reflection_options(assoc_value, options)
+        reflection_options = options.dup
+        reflection_options[:key] = polymorphic_key(assoc_value) if options[:polymorphic] && !options.key?(:key)
```
Comment (beauby) [inline lib/active_model/serializer/reflection.rb]: Note: We might want to make it a `||=` in order not to override user-provided options.

**Comment by beauby** [comment]
After giving it a thought, I'm not sure this feature really makes sense.

**Comment by beauby** [comment]
Closing, this does not make any sense.

**Pull request closed without merging**